openspoor/spoortakmodel/spoortak_model_mapper.py

```diff
@@ -0,0 +1,169 @@
+            return None
+
+        return self._data.models[model_version].loc[spoortak_identifier]
+
+    def _related_spoortakken(self, spoortak_identifier: str, geocodes: [int]) -> [(str, int)]:
+
+        related_spoortakken = []
+
+        for model_version in self._data.model_version_numbers:
+            temp_related_spoortakken = set()
+            model_changes = self._data.model_changes[model_version]
+            mask = (
+                    (model_changes['MODFWE'] == spoortak_identifier)
+                    | (model_changes['DASSIGNNAME'] == spoortak_identifier)
+                    | (model_changes['FWENAME'] == spoortak_identifier)
+            )
+
+            # not implemented
+            mask = mask & (model_changes['ACTION'] != 'RENAME')
+
+            changes = model_changes[mask]
+            if len(changes) > 0:
+                temp_related_spoortakken.update(changes['MODFWE'].unique())
+                temp_related_spoortakken.update(changes['DASSIGNNAME'].unique())
+                temp_related_spoortakken.update(changes['FWENAME'].unique())
+
+            if np.nan in temp_related_spoortakken:
+                temp_related_spoortakken.remove(np.nan)
+            if spoortak_identifier in temp_related_spoortakken:
+                temp_related_spoortakken.remove(spoortak_identifier)
+
+            for entry in temp_related_spoortakken:
+                # if there is no geocode overlap we can't assume the kilometrering uses the same 'kilometer lint'
+                spoortak = self._retrieve_spoortak(entry, model_version)
+                if spoortak is None:
+                    spoortak = self._retrieve_spoortak(entry, model_version - 1)
+                if spoortak is None:
+                    raise ValueError(
+                        f"Could not find spoortak {entry} in model {model_version} and {model_version - 1}")
+
+                if spoortak['GEOCODE_BEGIN'] in geocodes or spoortak['GEOCODE_EIND'] in geocodes:
+                    related_spoortakken.append((entry, model_version))
+
+        return related_spoortakken
+
+    @staticmethod
+    def _limit_start_end(subsections: [SpoortakSubsection], km_start: int, km_end: int) -> [SpoortakSubsection]:
+        """ perforce the limit in place """
+        return [subsection.limit_start_end(km_start, km_end) for subsection in subsections]
+
+    @staticmethod
+    def _remove_duplicates(subsections: [SpoortakSubsection]) -> [SpoortakSubsection]:
+        deduped = set(subsections)
+        return list(deduped)
+
+    def map(self, spoortak_subsection: SpoortakSubsection, _ignore_list: [str] = None) -> [SpoortakSubsection]:
+        """ Maps a spoortak subsection to all other spoortak models
+
+        :param spoortak_subsection: subsection to map
+        :param _ignore_list: list of spoortak identifiers to ignore (need this to avoid infinite loops)
+
+        """
+
+        if not _ignore_list:
```

Comment:
Can we set the default value of _ignore_list to [] directly instead?

Reply:
That would be risky, because an empty list [] is a mutable object.
if your function manipulates that empty list and say adds 'hello world'..
Then the second call would not have as default parameter [] but instead ['hello world] and that's unintended behavior.

And the resulting bugs are sometimes quite hard to figure out :-)